backend/backend/views.py

```diff
@@ -12,16 +15,26 @@
 from libs.user_login_throttle import UserLoginThrottle
 from substrapp.orchestrator import get_orchestrator_client
 from substrapp.utils import get_owner
-from users.utils import bearer_token as bearer_token_utils
+from users.models.token import BearerToken
 
 
 def _bearer_token_dict(token: BearerToken, include_payload: bool = True) -> dict:
-    d = {"created_at": token.created, "expires_at": bearer_token_utils.expires_at(token)}
+    d = {"created_at": token.created, "expires_at": token.expiry, "note": token.note, "token_id": token.token_id}
     if include_payload:
         d["token"] = token.key
     return d
 
 
+def _create_token_for_user(user, params) -> BearerToken:
+    if "expiry" not in params or "note" not in params:
+        raise ValidationError("expiry or note field are missing!")
+    return BearerToken.objects.create(
+        user=user,
+        expiry=None if params.get("expiry") == "never" else params.get("expiry"),
```

Comment:
This validation seems insufficient, what if the user send "toto" ? This is nether `None` or a `DateTime`